applications/tari_base_node/src/builder.rs

```diff
@@ -473,18 +471,60 @@ where
         .set_base_node_public_key(base_node_public_key.clone())
         .await
         .expect("Problem setting local base node public key for transaction service.");
+    let state_machine = base_node_handles
+        .get_handle::<StateMachineHandle>()
+        .expect("Could not get State Machine handle");
     let mut oms_handle = wallet_handles
         .get_handle::<OutputManagerHandle>()
         .expect("OutputManagerService is not registered");
-    oms_handle
-        .set_base_node_public_key(base_node_public_key)
-        .await
-        .expect("Problem setting local base node public key for output manager service.");
-    // Start the Output Manager UTXO Validation
-    oms_handle
-        .validate_utxos(UtxoValidationRetry::UntilSuccess)
-        .await
-        .expect("Problem starting the Output Manager Service Utxo Valdation process");
+    task::spawn(async move {
+        let mut status_watch = state_machine.get_status_info_watch();
+        let mut once_only = true;
+        loop {
+            let bootstrapped = match status_watch.recv().await {
+                Some(s) => match s {
+                    _ => s.bootstrapped,
+                },
+                None => false,
+            };
+            if bootstrapped {
+                debug!(
+                    target: LOG_TARGET,
+                    "Initial sync achieved and starting with UTXO validation process.",
+                );
+                oms_handle
+                    .set_base_node_public_key(base_node_public_key)
+                    .await
+                    .map_err(|e| {
+                        error!(
+                            target: LOG_TARGET,
+                            "An error occurred with Output Manager Service setting the base node public key: {}", e
+                        );
+                        e
+                    })
+                    .expect("An error occurred with Output Manager Service setting the base node public key.");
```

Comment:
I am concerned about a panic here, feels unnecessarily dangerous to have tasks panic. So I think log the error and then squash it.